zerver/lib/narrow.py

```diff
@@ -371,8 +381,10 @@ def by_in(self, query: Select, operand: str, maybe_negate: ConditionTransform) -
         assert self.user_profile is not None
 
         if operand == "home":
-            conditions = exclude_muting_conditions(self.user_profile, [])
-            return query.where(and_(*conditions))
+            # The `-is:muted` term is an alias for `in:home`. The `in:home` term will
+            # be removed in the future.
+            conditions = exclude_muting_conditions(self.user_profile, self.channel_data)
+            return query.where(maybe_negate(and_(*conditions)))
```

Comment:
It looks like there's a few independent changes here, like adding `maybe_negate` on this clause. Can you make each of those its own commit, with clear test updates in each? That'll make review a lot simpler.

Reply:
The existing `exclude_muting_conditions` logic is flawed, so `maybe_negate` will not work independently and I think its best to keep it with the rest of the changes to `exclude_muting_conditions`.

But I will split the commit into two commits, one for the new logic and another adding the `is:muted` alias.